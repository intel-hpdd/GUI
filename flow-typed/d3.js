// @flow

declare module d3 {
  declare module.exports: {
    max(xs:number[]):number;
    extent(xs:number[]):number[];
    scale:Object;
    time:Object;
  }
}
