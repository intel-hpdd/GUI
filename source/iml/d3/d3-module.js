// @flow

//
// Copyright (c) 2017 Intel Corporation. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import angular from 'angular';
import d3 from 'd3';

export default angular.module('d3', []).value('d3', d3).name;
