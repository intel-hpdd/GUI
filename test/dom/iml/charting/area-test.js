// @flow

import Inferno from 'inferno';

import Area from '../../../../source/iml/charting/area.js';
import d3 from 'd3';

it('should render as expected', () => {
  const svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
  const chartingGroup = document.createElementNS(
    'http://www.w3.org/2000/svg',
    'g'
  );
  svg.appendChild(chartingGroup);

  Inferno.render(
    <Area
      xScale={d3.scale.linear().domain([0, 100]).range([0, 200])}
      yScale={d3.scale.linear().domain([0, 100]).range([200, 0])}
      xValue={x => x}
      y1Value={x => x}
      chartingGroup={d3.select(chartingGroup).datum([0, 100])}
      color={'green'}
    />,
    svg
  );

  expect(svg).toMatchSnapshot();
});
