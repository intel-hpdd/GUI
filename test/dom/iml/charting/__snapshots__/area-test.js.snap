// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as expected 1`] = `
<svg>
  <g>
    <defs
      class="area-clip1"
    >
      <clippath
        id="area-clip1"
      >
        <rect
          width="200"
          height="200"
        />
      </clippath>
    </defs>
    <g
      clip-path="url(about:blank#area-clip1)"
      class="clipPath1"
    >
      <path
        class="area area1"
        stroke="green"
        fill="green"
        fill-opacity="0.2"
        d="M0,200L200,0L200,200L0,200Z"
      />
    </g>
  </g>
</svg>
`;
